### ISSUE TITLE
In the node replacement test - adding three nodes instead of one node

### ISSUE THREAD

**Issue by yitzhak12** [red-hat-storage/ocs-ci#6380]
In the test https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-deploy-ocs-cluster-prod/5016/testReport/tests.manage.z_cluster.nodes.test_node_replacement_proactive/TestNodeReplacementTwice/test_nodereplacement_twice/, it failed because it expects to have 6 worker nodes but instead find 8 worker nodes. When generating the new worker nodes, it added three nodes instead of one node.

**Comment by yitzhak12** [comment]
At the end of the test, Ceph Health was OK, so I don't think it's a product bug. Just need to investigate the reason it generated three nodes, even though it expected to create one.

**Comment by yitzhak12** [comment]
At the beginning of the test, it started with the worker nodes: `['compute-1', 'compute-2', 'compute-3', 'compute-4', 'compute-5', 'compute-6']`, then it deleted the node `compute-1`, and at the end of the test there were 8 worker nodes as you can see here: http://magna002.ceph.redhat.com/ocsci-jenkins/openshift-clusters/j-194vuf1cs36-t4a/j-194vuf1cs36-t4a_20220810T053529/logs/failed_testcase_ocs_logs_1660112973/test_nodereplacement_twice_ocs_logs/ocs_must_gather/quay-io-rhceph-dev-ocs-must-gather-sha256-4c8b41864994317237635a4b578d8649bb9a639013c597f968e81b43e119a7a9/cluster-scoped-resources/oc_output/get_nodes_-o_wide_--show-labels. somehow it added also the worker nodes: `compute-1, compute-2`. Even though the expectation was that it would add only the worker node `compute-7`. In the console output, we can see that we had 5 worker nodes, and it expects to add one more: 

> 2022-08-10 10:33:14  07:33:12 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: oc get Node  --selector=node-role.kubernetes.io/worker -o yaml
2022-08-10 10:33:14  07:33:14 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: oc get node  -o yaml
2022-08-10 10:33:14  07:33:14 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: oc get Node compute-2
2022-08-10 10:33:14  07:33:14 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: oc get Node  -o yaml
2022-08-10 10:33:15  07:33:15 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: oc get Node compute-3
2022-08-10 10:33:15  07:33:15 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: oc get Node  -o yaml
2022-08-10 10:33:15  07:33:15 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: oc get Node compute-4
2022-08-10 10:33:16  07:33:15 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: oc get Node  -o yaml
2022-08-10 10:33:16  07:33:16 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: oc get Node compute-5
2022-08-10 10:33:16  07:33:16 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: oc get Node  -o yaml
2022-08-10 10:33:16  07:33:16 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: oc get Node compute-6
2022-08-10 10:33:17  07:33:16 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: oc get Node  -o yaml
2022-08-10 10:33:17  07:33:17 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: oc get Node control-plane-0
2022-08-10 10:33:17  07:33:17 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: oc get Node  -o yaml
2022-08-10 10:33:17  07:33:17 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: oc get Node control-plane-1
2022-08-10 10:33:18  07:33:17 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: oc get Node  -o yaml
2022-08-10 10:33:18  07:33:18 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: oc get Node control-plane-2
2022-08-10 10:33:18  07:33:18 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: oc get Node  -o yaml
2022-08-10 10:33:18  07:33:18 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: oc get clusterversion -n openshift-storage -o yaml
2022-08-10 10:33:19  07:33:18 - MainThread - ocs_ci.ocs.platform_nodes - INFO  - Adding Nodes of type RHCOS
2022-08-10 10:33:19  07:33:18 - MainThread - ocs_ci.ocs.platform_nodes - INFO  - Existing worker nodes: 5, New nodes to add: 1
2022-08-10 10:33:19  07:33:18 - MainThread - ocs_ci.utility.csr - INFO  - Retrieving CSR data

I would suggest that the problem occurs in the function `add_node` https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/ocs/platform_nodes.py#L1616. When generating the new CSRs, it generated two more CSRs

**Comment by yitzhak12** [comment]
@vavuthu @petr-balogh Please let me know if there were any recent changes in the function add_node https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/ocs/platform_nodes.py#L1616, that may cause this issue. Or if you have any idea why it happened.

**Comment by petr-balogh** [comment]
I am not aware of any recent change here. @vavuthu are you looking at the issue? I see you self assigned it to yourself.

**Comment by yitzhak12** [comment]
Maybe the pr https://github.com/red-hat-storage/ocs-ci/pull/6378 and the recent changes from the pr https://github.com/red-hat-storage/ocs-ci/pull/6387 will fix the issue.

**Comment by vavuthu** [comment]
@yitzhak12 Could you try with above chnages. It looks like deleted ones compute-0 and others node came back to life.

**Comment by yitzhak12** [comment]
Yes, the pr #6387 is merged, so we will see if it works fine in the following weekend's test runs.

**Comment by am-agrawa** [comment]
> Yes, the pr #6378 is merged, so we will see if it works fine in the following weekend's test runs.

@yitzhak12 Could you pls look at the subsequent runs which happened after pr #6378 got merged so that we can close this issue?
